snapcraft/commands/clean.py

```diff
@@ -39,6 +41,34 @@
 logger = logging.getLogger(__name__)
 
 
+def _reverse_dependency_tree(config, part_name):
+    dependents = config.part_dependents(part_name)
+    for dependent in dependents.copy():
+        dependents |= _reverse_dependency_tree(config, dependent)
```

Comment:
a comment that this is never circular because the validation of the yaml forbids it will be handy for future developers looking at this code.